### PR TITLE
Add doc for @link

### PR DESCRIPTION
**Pull request by mkotsbak**
Until there is a separate documentation page about writing facades, please add something about @link.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -38,6 +38,9 @@ updates.](http://twitter.com/scala_native)
 
   val ptr = stdlib.malloc(32)
   ```
+  
+  Use the @link("libraryname") annotation to automatically link with
+  a native library whenever an annotated extern object is used. (e.g. @link("png") to link with "libpng")
```
Comment (sjrd) [inline README.md]: You should surround code snippets with backticks (`) for proper formatting.

Comment (mkotsbak) [inline README.md]: Sure. Better now?

Comment (sjrd) [inline README.md]: Yep. I'll leave the final call to @densh, though.

**Comment by densh** [comment]
Hi @mkotsbak, readme was never meant to be an exhaustive doc for all of the functionality. It's just a show-case of what's interesting. Lets put this somewhere else in `doc/` folder for now.

**Comment by mkotsbak** [comment]
See #297 instead.

**Pull request closed without merging**